pkg/server/admin.go

```diff
@@ -301,8 +301,10 @@ func (s *adminServer) AllMetricMetadata(
 ) (*serverpb.MetricMetadataResponse, error) {
 
 	md, _, _ := s.metricsRecorder.GetMetricsMetadata(true /* combine */)
+	metricNames := s.metricsRecorder.GetRecordedMetricNames(md)
```

Comment:
I might be oversimplifying this, but why couldn't `GetMetricsMetadata` include the recorded names in its payload?

Is it possible to update [the Metadata definition](https://github.com/cockroachdb/cockroach/blob/16d37601e86d03a19fd0aa6413910a59100462d8/pkg/util/metric/metric.proto#L52) so that this information could be included?

Reply:
I don't think it makes sense to include this in Metadata definition because it the recorded name / prefix is a related to the registry in which a the metric is written to and not the metric itself. See https://github.com/cockroachdb/cockroach/blob/16d37601e86d03a19fd0aa6413910a59100462d8/pkg/server/status/recorder.go#L397. This would also require those who define new metrics to know about how the metric is being recorded to tsdb.

Reply:
Ahhh I see, that makes a lot more sense, thanks 👍